models/gif_psc_exp.cpp

```diff
@@ -0,0 +1,486 @@
+/*
+ *  gif_psc_exp.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "gif_psc_exp.h"
+
+// C++ includes:
+#include <limits>
+
+// Includes from nestkernel:
+#include "exceptions.h"
+#include "kernel_manager.h"
+#include "universal_data_logger_impl.h"
+
+// Includes from sli:
+#include "dict.h"
+#include "integerdatum.h"
+#include "doubledatum.h"
+#include "dictutils.h"
+
+#include "numerics.h"
+#include "compose.hpp"
+#include "propagator_stability.h"
+
+namespace nest
+{
+/* ----------------------------------------------------------------
+ * Recordables map
+ * ---------------------------------------------------------------- */
+
+RecordablesMap< gif_psc_exp > gif_psc_exp::recordablesMap_;
+
+// Override the create() method with one call to RecordablesMap::insert_()
+// for each quantity to be recorded.
+template <>
+void
+RecordablesMap< gif_psc_exp >::create()
+{
+  // use standard names whereever you can for consistency!
+  insert_( names::V_m, &gif_psc_exp::get_V_m_ );
+  insert_( names::E_sfa, &gif_psc_exp::get_E_sfa_ );
+  insert_( names::stc, &gif_psc_exp::get_stc_ );
+  insert_( names::input_currents_ex, &gif_psc_exp::get_input_currents_ex_ );
+  insert_( names::input_currents_in, &gif_psc_exp::get_input_currents_in_ );
+}
+
+/* ----------------------------------------------------------------
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::gif_psc_exp::Parameters_::Parameters_()
+  : g_L_( 4.0 )        // nS
+  , E_L_( -70.0 )      // mV
+  , c_m_( 80.0 )       // pF
+  , V_reset_( -55.0 )  // mV
+  , Delta_V_( 0.5 )    // mV
+  , V_T_star_( -35 )   // mV
+  , lambda_0_( 0.001 ) // 1/ms
+  , I_e_( 0.0 )        // pA
+  , t_ref_( 4.0 )      // ms
+  , tau_ex_( 2.0 )     // ms
+  , tau_in_( 2.0 )     // ms
+  , tau_sfa_()         // ms
+  , q_sfa_()           // mV
+  , tau_stc_()         // ms
+  , q_stc_()           // nA
```

Comment:
*All* members should be initialized, and their order here must (should) be the same as the order of declarations in the header file, because C++ initializes values in the order of declaration, *not* in the order in which initializers are listed in the constructor.